src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -2008,6 +2007,14 @@ else if (part.matches("\\d+")) {
             }
         }
 
+        // If found Abbreviation, make the next letter lowercase
+        for (int index = 0; index < extractedRuleName.length() - 1; index++) {
+            if (Character.isUpperCase(extractedRuleName.charAt(index))) {
+                extractedRuleName.setCharAt(
+                        index + 1, Character.toLowerCase(extractedRuleName.charAt(index + 1)));
+            }
```

Comment:
I have worries on this.
Lets do as I suggested simple removal of " & " from test to make words around be single word.
"K & R" --> "KR" , you should do this after `ruleNameWithoutPrecedingNumbers.split`

Reply:
> "K & R" --> "KR" , you should do this after ruleNameWithoutPrecedingNumbers.split

We're already following this. After `ruleNameWithoutPrecedingNumbers.split` we get an array of words. Then we make the first letter of the word capital and other letters to lower case.

Here after removing the `&`, we get an array of words where `K` and `R` both are considered as different words. After that we combine `K`, `R` to `KR` and then it gets converted to `Kr`.

I think this logic is fine there should not be any problem with this.

Reply:
> "K & R" --> "KR" , you should do this after ruleNameWithoutPrecedingNumbers.split

We're already following this. After `ruleNameWithoutPrecedingNumbers.split` we get an array of words. Then we make the first letter of the word capital and other letters to lower case in foreach loop. Here the special symbols gets ignored so the `&` will get ignored.

In the array of words `K` and `R` both are considered as different words. After that we combine `K`, `R` to `KR` in the foreach loop and then it gets converted to `Kr` in the last loop.

I think this logic is fine there should not be any problem with this.

Reply:
So you're suggesting to combine `K` and `R` into one word `KR` before `for (String part : parts)` this means that we're combing all the words into a single string like `NonemptyblocksKRstyle`. Then we cannot split the string or perform the camel case operation. 

If we write some logic for whenever we encounter a capital letter, we make the next letter lower case then it'll not work for: `Non-ASCII characters`

Reply:
update is pushed